ng2-material/components/paging/paging.ts

```diff
@@ -0,0 +1,286 @@
+        <button (click)="nextPage()" class="material-icons">keyboard_arrow_right</button>
+      </span>
+    `,
+  host: {
+    '[class.md-paging-controls]': 'true'
+  }
+})
+export class MdPagingControls {
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPagingModel;
+
+  isFirstPage() {
+    return isPresent(this.model) && this.model.currentPage == 1;
+  }
+
+  isLastPage() {
+    return isPresent(this.model) && this.model.totalItems <= this.model.currentPage * this.model.itemsPerPage;
+  }
+
+  previousPage() {
+    this._changePage(this.model.currentPage - 1);
+  }
+
+  nextPage() {
+    this._changePage(this.model.currentPage + 1);
+  }
+
+  _changePage(newPage: number) {
+    let _model = JSON.parse(JSON.stringify(this.model));
+    _model.currentPage = newPage;
+    this._service.change(this.name, _model);
+  }
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        this.model = event.paging;
+      });
+  }
+}
+
+@Component({
+  selector: 'md-paging-length-selector',
+  template: `
+   <span *ngIf="selectorLengthBefore" class="md-paging-length-selector-label md-paging-length-selector-before">{{selectorLengthBefore}}</span>
+   <select [(ngModel)]="model.itemsPerPage" (ngModelChange)="changePagingLength($event)" class="md-paging-length-select">
+      <option *ngFor="#length of selectorLengthAvailable" [value]="length">
+        {{length}}
+      </option>
+    </select>
+    <span *ngIf="selectorLengthAfter" class="md-paging-length-selector-label md-paging-length-selector-after">{{selectorLengthAfter}}</span>
+  `,
+  host: {
+    '[class.md-paging-length-selector]': 'true',
+    '[hidden]': '!selectorLengthAvailable.length'
+  }
+})
+export class MdPagingLengthSelector {
+  @Input()
+  selectorLengthBefore: string = 'Rows per page:';
+
+  @Input()
+  selectorLengthAfter: string;
+
+  @Input()
+  selectorLengthAvailable: Array<number> = [];
+
+  @Input()
+  name: string = 'default';
+
+  @Input()
+  model: IPagingModel = {
+    currentPage: 0,
+    itemsPerPage: 0,
+    totalItems: 0
+  };
+
+  constructor(private _service: PagingService) {
+    this._service.onChange
+      .filter(event => isPresent(event) && isPresent(event.name))
+      .filter(event => event.target === this.name)
+      .subscribe(event => {
+        if (isPresent(event.paging)) {
+          this.model = event.paging;
+        }
+      });
+  }
+
+  changePagingLength(value) {
+    let _model = JSON.parse(JSON.stringify(this.model));
+    _model.currentPage = 1;
+    _model.itemsPerPage = parseInt(value);
+    this._service.change(this.name, _model);
+  }
+
+}
+
+export interface IPagingChange {
+  name: string;
+  target: string;
+  paging: IPagingModel;
+}
+
+@Component({
+  selector: 'md-paging',
```

Comment:
personally prefer "noun" component names to "verb" since components are things (not just behaviors) - so `md-pager` selector and `MdPager` class name would be clear to me.